Makefile

```diff
@@ -320,6 +343,32 @@ release-build:
 	cd hack && $(GO) run release.go --templateDir ./templates/values/ --outputDir ../deployment/network-operator/
 	cd hack && $(GO) run release.go --templateDir ./templates/config/manager --outputDir ../config/manager/
 
+# dev environment
+
+MINIKUBE_CLUSTER_NAME = net-op-dev
+dev-minikube: $(MINIKUBE) ## Create a minikube cluster for development.
+	CLUSTER_NAME=$(MINIKUBE_CLUSTER_NAME) MINIKUBE_BIN=$(MINIKUBE) $(CURDIR)/hack/scripts/minikube-install.sh
+
+clean-minikube: $(MINIKUBE)  ## Delete the development minikube cluster.
+	$(MINIKUBE) delete -p $(MINIKUBE_CLUSTER_NAME)
+
+
+CERT_MANAGER_YAML=$(MANIFESTDIR)/cert-manager.yaml
+$(CERT_MANAGER_YAML): | $(MANIFESTDIR)
+	curl -fSsL "https://github.com/cert-manager/cert-manager/releases/download/v1.13.3/cert-manager.yaml" -o $(CERT_MANAGER_YAML)
+
+SKAFFOLD_REGISTRY=localhost:5000
+dev-skaffold: $(SKAFFOLD) $(CERT_MANAGER_YAML) dev-minikube ## Create a development minikube cluster and deploy the operator in debug mode.
+	## Deploy the network attachment definition CRD.
+	kubectl apply -f hack/crds/*
+	## Deploy cert manager to provide certificates for webhooks.
+	kubectl apply -f $(CERT_MANAGER_YAML) \
+	&& $Q echo "Waiting for cert-manager deployment to be ready."\
+	&& kubectl wait --for=condition=ready pod -l app=webhook --timeout=60s -n cert-manager
+	# Use minikube for docker build and deployment.
+	$Q eval $$($(MINIKUBE) -p $(MINIKUBE_CLUSTER_NAME) docker-env); \
+	$(SKAFFOLD) debug --default-repo=$(SKAFFOLD_REGISTRY) --detect-minikube=false
```

Comment:
I have a plan to modify this target to make it possible to alternatively use an existing cluster by passing a kubeconfig. Will do that in a follow-up PR though.